internal/commands/accesslist/delete.go

```diff
@@ -0,0 +1,195 @@
+type CommandDelete struct {
+	inputs deleteInputs
+}
+
+// Flags is the command flags
+func (cmd *CommandDelete) Flags() []flags.Flag {
+	return []flags.Flag{
+		cli.AppFlagWithContext(&cmd.inputs.App, "to remove an entry in its Access List"),
+		cli.ProjectFlag(&cmd.inputs.Project),
+		cli.ProductFlag(&cmd.inputs.Products),
+		flags.StringSliceFlag{
+			Value: &cmd.inputs.Addresses,
+			Meta: flags.Meta{
+				Name: "ip",
+				Usage: flags.Usage{
+					Description: "Specify the IP address or CIDR block that you would like to delete",
+				},
+			},
+		},
+	}
+}
+
+// Inputs is the command inputs
+func (cmd *CommandDelete) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+// Handler is the command handler
+func (cmd *CommandDelete) Handler(profile *user.Profile, ui terminal.UI, clients cli.Clients) error {
+	app, err := cli.ResolveApp(ui, clients.Realm, cmd.inputs.Filter())
+	if err != nil {
+		return err
+	}
+
+	allowedIPs, err := clients.Realm.AllowedIPs(app.GroupID, app.ID)
+	if err != nil {
+		return err
+	}
+
+	selected, err := cmd.inputs.resolveAllowedIP(ui, allowedIPs)
+	if err != nil {
+		return err
+	}
+
+	if len(selected) == 0 {
+		ui.Print(terminal.NewTextLog("No IP addresses or CIDR blocks to delete"))
+		return nil
+	}
+
+	outputs := make(deleteAllowedIPOutputs, len(selected))
+	for i, allowedIP := range selected {
+		err := clients.Realm.AllowedIPDelete(app.GroupID, app.ID, allowedIP.ID)
+		outputs[i] = deleteAllowedIPOutput{allowedIP, err}
+	}
+
+	sort.SliceStable(outputs, func(i, j int) bool {
+		return outputs[i].err != nil && outputs[j].err == nil
+	})
+
+	tableRows := make([]map[string]interface{}, 0, len(outputs))
+	for _, output := range outputs {
+		row := map[string]interface{}{
+			headerAddress: output.allowedIP.Address,
+			headerComment: output.allowedIP.Comment,
+		}
+		tableRows = append(tableRows, row)
+		tableRowDelete(output, row)
+	}
+
+	ui.Print(terminal.NewTableLog(
+		fmt.Sprintf("Deleted %d IP address(es) and CIDR block(s)", len(outputs)),
+		deleteTableHeaders,
+		tableRows...,
+	))
+
+	return nil
+}
+
+func (i *deleteInputs) Resolve(profile *user.Profile, ui terminal.UI) error {
+	if err := i.ProjectInputs.Resolve(ui, profile.WorkingDirectory, false); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (i *deleteInputs) resolveAllowedIP(ui terminal.UI, allowedIPs []realm.AllowedIP) ([]realm.AllowedIP, error) {
+	if len(allowedIPs) == 0 {
+		return nil, nil
+	}
+
+	if len(i.Addresses) > 0 {
+		allowedIPsByAddress := make(map[string]realm.AllowedIP, len(allowedIPs))
+		for _, allowedIP := range allowedIPs {
+			allowedIPsByAddress[allowedIP.Address] = allowedIP
+		}
+
+		allowedIPs := make([]realm.AllowedIP, 0, len(i.Addresses))
+		for _, identifier := range i.Addresses {
+			if allowedIP, ok := allowedIPsByAddress[identifier]; ok {
+				allowedIPs = append(allowedIPs, allowedIP)
+			}
+		}
+
+		if len(allowedIPs) == 0 {
+			return nil, errors.New("unable to find allowed IPs")
```

Comment:
and then depending on how you handle the cases where specified addresses aren't found, this check could seemingly go away (and then be handled similarly above in your `Handler` code where you call:

`ui.Print(terminal.NewTextLog("No IP addresses or CIDR blocks to delete"))`